apps/organisations/views.py

```diff
@@ -215,44 +218,76 @@ def get_context_data(self, **kwargs):
 
     def form_valid(self, form):
         data = form.cleaned_data
-        img = self.generate_image(data)
         context = self.get_context_data()
+        img = self.generate_image(data)
+        # get the content form with unchanged cleaned data, as image generation
+        # changes the image data (why?)
+        content_form = CommunicationContentCreationForm(
+            initial=data, project=self.project, format=self.format)
         context['image_preview'] = img
+        context['content_form'] = content_form
         return self.render_to_response(context)
 
     def generate_image(self, data):
-        # Dummy images
-        image = Image.new('RGB', (304, 192), color='red')
-        logo1 = Image.new('RGB', (100, 10), color='blue')
-        logo2 = Image.new('RGB', (100, 10), color='green')
-
-        # Adding padding
+        # Get required image size and image
+        image_format = SOCIAL_MEDIA_SIZES[self.format]
+        image_width = image_format['img_min_width']
+        image_height = image_format['img_min_height']
+        image_get = Image.open(data['image'].file)
+        image_size = (image_width, image_height)
+        image = image_get.resize(image_size)
+
+        # Get organisations logo
+        logo_org_get = Image.open(self.organisation.logo)
+        logo_org_size = (144, 144)
+        logo_org = logo_org_get.resize(logo_org_size)
+
+        # Get aplus logo
+        logo_aplus_get = Image.open(
+            os.path.join(
+                settings.STATIC_ROOT,
+                'images/logo.png'))
```

Comment:
This doesnt work for me as image is in assets instead of static?

Reply:
@Rineee sorry you need to run `python manage.py collectstatic` or probably `make release` does it too

Reply:
ah ups, sorry!

Reply:
we could also change the path to assets here so you don't have to run collectstatic to make your dev environment work (or do we need to do this anyway for other parts?)